pydantic/fields.py

```diff
@@ -321,6 +321,14 @@ class MyModel(pydantic.BaseModel):
                 spam: Annotated[int, pydantic.Field(gt=4)] = 4  # <-- or this
             ```
         """
+
+        if annotation is default:
+            raise RuntimeError(
```

Comment:
@Viicos,

Circling back to this solution - I do think we should go ahead and implement something along the lines of what you have here. Could you use a PydanticUserError here instead, as Samuel mentioned?

Happy to review once you make that change 👍

Reply:
Yes I think for now this would be the best we can do. It doesn't cover all cases (e.g. [this](https://github.com/pydantic/pydantic/pull/8243#issuecomment-1830485561) and [this](https://github.com/pydantic/pydantic/pull/8243#issuecomment-1830510003)) but this is still better than nothing. Will apply changes soon

Reply:
Thanks! Yep, let's stick with this for now, then can consider a more complex solution in the future.